test/unit/host_test.rb

```diff
@@ -2186,6 +2187,7 @@ def to_managed!
   end
 
   test '#primary_interface works during deletion' do
+    skip 'Waiting for rails4 '
```

Comment:
This test actually tests that ActiveRecord's caching on relations keeps instances of all interfaces, even if they are deleted. 
I had to add `after_save :clear_association_cache` to our host model in order to fix references issue after save ( `host.some_aspect.host.object_id` is not equal to `host.object_id`). 
In Rails 4 the issue with the relations is fixed, so then it will be logical to remove the `after_save` hook and put the test back.